vaultlocker/exceptions.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+class VaultlockerException(Exception):
+
+    def __init__(self, *args):
+        if args:
+            self.message = args[0]
+        else:
+            self.message = "Empty VaultlockerException"
+
+    def __str__(self):
+        return self.message
+
+
+class VaultWriteError(VaultlockerException):
+
+    def __init__(self, path, error):
+        self.path = path
+        self.error = error
+
+    def __str__(self):
```

Comment:
Let's make use of the behavior (`__init__` and `__str__`) you implemented in `VaultlockerException` already. I suggest this (not tested but should work):

```
class VaultWriteError(VaultlockerException):

    def __init__(self, path, error):
        super().__init__("Can't read vault at path {}, error: {}".format(
￼           path, error))

# That's it! No need for self.path, self.error, __str__
```

By doing this we are leveraging the parent implementation of `__str__`. What do you think?